var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -90,28 +91,37 @@ class PyNumpy(PythonPackage):
     variant("blas", default=True, description="Build with BLAS support")
     variant("lapack", default=True, description="Build with LAPACK support")
 
-    # Based on wheel availability on PyPI
-    depends_on("python@3.9:3.11", when="@1.25:", type=("build", "link", "run"))
+    depends_on("python@3.9:3.12", when="@2.26:", type=("build", "link", "run"))
+    depends_on("python@3.9:3.11", when="@1.25", type=("build", "link", "run"))
     depends_on("python@3.8:3.11", when="@1.23.2:1.24", type=("build", "link", "run"))
     depends_on("python@3.8:3.10", when="@1.22:1.23.1", type=("build", "link", "run"))
     depends_on("python@:3.10", when="@1.21.2:1.21", type=("build", "link", "run"))
     depends_on("python@:3.9", when="@1.19.3:1.21.1", type=("build", "link", "run"))
     depends_on("python@:3.8", when="@1.17.3:1.19.2", type=("build", "link", "run"))
     depends_on("python@:3.7", when="@1.14.5:1.17.2", type=("build", "link", "run"))
 
-    # https://github.com/spack/spack/pull/32078
-    depends_on("py-setuptools@:63", type=("build", "run"))
-    depends_on("py-setuptools@:59", when="@:1.22.1", type=("build", "run"))
-    # Check pyproject.toml for updates to the required cython version
+    depends_on("py-cython@0.29.34:3.0", when="@1.26:", type="build")
     depends_on("py-cython@0.29.34:2", when="@1.25:", type="build")
     depends_on("py-cython@0.29.13:2", when="@1.18.0:", type="build")
     depends_on("py-cython@0.29.14:2", when="@1.18.1:", type="build")
     depends_on("py-cython@0.29.21:2", when="@1.19.1:", type="build")
     depends_on("py-cython@0.29.24:2", when="@1.21.2:", type="build")
     depends_on("py-cython@0.29.30:2", when="@1.22.4:", type="build")
+    depends_on("py-pyproject-metadata@0.7.1:", when="@1.26:", type="build")
+    depends_on("py-tomli@1:", when="@1.26: ^python@:3.10", type="build")
+    depends_on("py-setuptools@60:", when="@1.26: ^python@3.12:", type="build")
+    # https://github.com/spack/spack/pull/32078
+    depends_on("py-setuptools@:63", when="@:1.25", type=("build", "run"))
+    depends_on("py-setuptools@:59", when="@:1.22.1", type=("build", "run"))
+    depends_on("py-colorama", when="@1.26: platform=windows", type="build")
+
+    # meson is vendored, ninja and pkgconfig are not
```

Comment:
Out of curiosity, why was meson vendored?

Reply:
Because we needed the SIMD support functionality which hasn't been merged into Meson proper yet.